feat: Create User Button in Header with Dropdown Menu

johnnyzaet08 opened the pull request:
# 📝  feat: Create User Button in Header with Dropdown Menu

## 🛠️ Issue
- Closes #61 

## 📖 Description
- The `dropdown-menu.tsx` component has been added, sourced from the official shadcn/ui documentation, to enable a user dropdown menu in the application’s header. To support this component, the `@radix-ui/react-dropdown-menu` dependency was installed. In addition, the `app/marketplace/page.tsx` file was updated to include the necessary imports for both the new dropdown menu and related icons, as well as the required JSX structure to display the user menu options. This integration now provides a cleaner, more intuitive way for users to access their profile, listings, transaction history, and settings directly from the header.

## ✅ Changes made
- Added the `dropdown-menu.tsx` component to the project.
- Installed and updated dependencies to ensure the dropdown functionality works as intended.
- Modified `app/marketplace/page.tsx` to incorporate the necessary imports and markup for the user dropdown menu.

## 🖼️ Media (screenshots/videos)
- The implementation of the User button is displayed
![Screenshot 2024-12-13 132752](https://github.com/user-attachments/assets/809311a6-1f52-4219-a8c7-dd5f685cf651)
- The dropdown is displayed when the user button is clicked.
![Screenshot 2024-12-13 132808](https://github.com/user-attachments/assets/9fc604f7-56a3-43e3-bf78-b8248ca133a5)
- The button is checked for responsive operation
![Screenshot 2024-12-13 132727](https://github.com/user-attachments/assets/0b02ca06-a04a-47cc-8028-2f008ef16a31)


## 📜 Additional Notes
- The implementation of another user within the header was taken into consideration, the structure of the user was respected and the new button was added within it.

derianrddev commented on frontend/app/marketplace/page.tsx:
```diff
@@ -312,10 +324,43 @@ function HeaderComponent({ searchTerm, setSearchTerm }: HeaderComponentProps) {
+						</Button>
+					</DropdownMenuTrigger>
+					<DropdownMenuContent className="w-48">
+						<DropdownMenuLabel>User Menu</DropdownMenuLabel>
```
Remove the `DropdownMenuLabel`

derianrddev commented on frontend/app/marketplace/page.tsx:
```diff
@@ -312,10 +324,43 @@ function HeaderComponent({ searchTerm, setSearchTerm }: HeaderComponentProps) {
+					</DropdownMenuTrigger>
+					<DropdownMenuContent className="w-48">
+						<DropdownMenuLabel>User Menu</DropdownMenuLabel>
+						<DropdownMenuSeparator/>
```
Remove the `DropdownMenuSeparator`

derianrddev commented:
@johnnyzaet08 Good job! Just some minor comments. Resolve the conflicts and add the user button to the new header, please.

johnnyzaet08 commented:
Hey @derianrddev, I need to wait for @danielcdz approval? Or can I do the merge right away? To solve the conflicts and make the merge.

derianrddev commented:
Hey @johnnyzaet08 , you can go ahead and do the merge right now. No problem.

danielcdz commented:
@johnnyzaet08 only maintainers can merge a PR! and this after receive at least 2 approvals on your PR

johnnyzaet08 commented:
Understood @danielcdz. 
Hey @derianrddev, I just did the update.  When I fixed the merge conflicts I got confused, then I realized and fixed it. Here is the test of the integration of the new button to the new header.

- Button integrated in the new header.
![Screenshot 2024-12-15 222600](https://github.com/user-attachments/assets/3e7c4292-e4b4-48a6-8bdf-ae14e76dac16)

derianrddev commented:
@johnnyzaet08 Thanks for the update, it looks good. We just need @danielcdz's approval to merge it into the project.

danielcdz commented:
@johnnyzaet08 one last update to fix the conflicts and then we can merge!
